File: infra/question/question.js
db = db.getSiblingDB("krakit-question");

if (!db.getCollectionNames().includes("questions")) {
  db.createCollection("questions", {
    validator: {
      $jsonSchema: {
        bsonType: "object",
        required: [
          "exam_id",
          "section_id",
          "question_set_id",
          "question_text",
          "type",
          "difficulty"
        ],
        properties: {
          exam_id: { bsonType: "string" },
          section_id: { bsonType: "string" },
          subject_id: { bsonType: "string" },

          question_set_id: { bsonType: "string" },
          passage_id: { bsonType: "objectId" },

          question_text: { bsonType: "string" },
          question_image_url: { bsonType: "string" },

          options: {
            bsonType: "array",
            items: {
              bsonType: "object",
              required: ["key"],
              properties: {
                key: { bsonType: "string" },
                option_text: { bsonType: "string" },
                option_image_url: { bsonType: "string" }
              }
            }
          },

          correct_answer: {},
          type: { enum: ["MCQ", "MSQ", "NUMERIC"] },

          marks: { bsonType: "number" },
          negative_marks: { bsonType: "number" },

          difficulty: { enum: ["easy", "medium", "hard"] },

          topic: { bsonType: "string" },
          is_active: { bsonType: "bool" },

          created_at: { bsonType: "date" }
        }
      }
    }
  });
}

// Indexes
db.questions.createIndex({ question_set_id: 1 });
db.questions.createIndex({ exam_id: 1, section_id: 1 });
db.questions.createIndex({ passage_id: 1 });
db.questions.createIndex({ difficulty: 1 });
db.questions.createIndex({ topic: 1 });

print("✅ questions ready");

File: services/exam-service/internal/dto/question.go
package dto

type Question struct {
	ID            string `json:"id" bson:"_id,omitempty"`
	ExamID        string `json:"exam_id"`
	SectionID     string `json:"section_id"`
	SubjectID     string `json:"subject_id,omitempty"`
	QuestionSetID string `json:"question_set_id"`

	PassageID *string `json:"passage_id,omitempty"`

	QuestionText  string `json:"question_text"`
	QuestionImage string `json:"question_image_url,omitempty"`

	Options []Option `json:"options"`

	CorrectAnswer interface{} `json:"correct_answer"`

	Type string `json:"type"` // MCQ, MSQ, NUMERIC

	Marks         float64 `json:"marks"`
	NegativeMarks float64 `json:"negative_marks"`

	Difficulty string `json:"difficulty"`

	Topic    string `json:"topic,omitempty"`
	IsActive bool   `json:"is_active"`
}

type QuestionWithMeta struct {
	Question
	SectionID string `json:"section_id"`
	Order     int    `json:"order"`
}
